### DISCUSSION TITLE
Disable mouse on DataTable?

### DISCUSSION THREAD

**Discussion by fohrloop** [Textualize/textual#5200] [Q&A] [answered]
I have subclassed the [DataTable](https://textual.textualize.io/widgets/data_table/) and defined two "modes"
- One for moving the cursor
- One for moving the selected row

The fact that user is able to click with mouse also on the "moving the selected item" mode makes the app not work intuitively. I don't think mouse support is crucial for me, especially with the DataTable component. It would be easier to disable mouse actions than try to add support for it. I would like to either disable the ability to click cells (and change selected cell) on DataTable selectively or completely. I tried to override DataTable methods (without really checking what they do):

```python
    async def _on_click(self, event) -> None: ...

    def action_select_cursor(self) -> None: ...

    def _on_mouse_move(self, event): ...
```

but that did not have any effect. I also tried:

```python
    async def on_event(self, event: events.Event) -> None:
        if isinstance(event, MouseEvent):
            event.prevent_default()
        await super().on_event(event)
```
mentioned [here](https://github.com/Textualize/textual/discussions/183) but that does not work.

Any ideas how mouse clicks could be disabled?

**Comment by TomJGooding** [comment, marked as answer]
You can use the `prevent_default` method as mentioned here: https://textual.textualize.io/guide/events/#preventing-default-behaviors

```python
    async def _on_click(self, event: events.Click) -> None:
        event.prevent_default()
```

**Reply by fohrloop** [reply]
wonderful! thanks!
